src/Areas/Server/Commands/TypeToJsonTypeMapper.cs

```diff
@@ -74,4 +82,76 @@ public static string ToJsonType(this Type? type)
 
         return "object";
     }
+
+    /// <summary>
+    /// Gets the element type of an array or collection type.
+    /// </summary>
+    /// <param name="type">The array or collection type to analyze.</param>
+    /// <returns>The element type if the type is an array or collection, otherwise null.</returns>
+    public static Type? GetArrayElementType(Type type)
+    {
+        ArgumentNullException.ThrowIfNull(type);
+
+        // Handle arrays
+        if (type.IsArray)
+        {
+            return type.GetElementType();
+        }
+
+        // Handle generic collections like List<T>, IEnumerable<T>, etc.
+        if (type.IsGenericType)
+        {
+            var genericArgs = type.GetGenericArguments();
+            if (genericArgs.Length == 1)
```

Comment:
I'm not an expert in C#, but in Java we can have generics with more than one element. What would happen if that were the case here?